DESIGN.md

```diff
@@ -0,0 +1,72 @@
+    * Define the style in `core/styles.xml`. For our default styles, applied with theme attributes, the naming convention is `Widget.KiwixTheme.*` and these should ideally inherit from their `Widget.MaterialComponent.*`, should no such parent exist, it is generally advisable to google "how to style [Component]". When there are two or more of a view that differ stylistically from our default styling, but are identical to each other visually, it is appropriate to extract a style to keep the XML DRY. A non-default style should use a descriptive name. e.x. `list_item_title`. 
+2. Add the color or style to a specific theme attribute in `themes.xml`.
+3. Make sure that the color or style works in both day and night mode. If it does not, add the dark mode compatible attribute to `values-night/themes.xml`.
+
+For a video on how to work with themes, styles and colors, see [Developing Themes with Style (Android Dev Summit '19)](https://www.youtube.com/watch?v=Owkf8DhAOSo).
+
+## Night Mode / Dark Theme
+Night mode is a different theme that has many benefits such as saving battery and improving visibility in low light conditions [2]. When doing any design, make sure the new design works in dark mode. Night mode can be activated from the systems settings or from the kiwix app. 
+
+Whenever a resource exists in both `res/*-night` and `res/*` it will be used as such. For example: `kiwix_icon_with_title.png` exists in both `res/drawable` and `res/drawable-night` which means that the image in `res/drawable-night` will be used automagically in night mode. For further reading on development with dark mode, see [Material Design - Dark Theme](https://developer.android.com/guide/topics/ui/look-and-feel/darktheme). 
+
+## Typography
+Text should contrast well, use the correct weight and size to present content as clearly and efficiently as possible. 
+
+Views using the same style often use different text style. Therefore text styles, and view styles are seperated and can be used simultaneously. To keep an android application consistent in style, thirteen text style attributes should be defined. These style attributes can be generated ([Material Design type scale generator](https://material.io/design/typography/the-type-system.html#type-scale)), or defined by editing `core/../values/type.xml`. 
+
+All text should have a text style to simplifiy appearance changes. To set a text style, simply set the `textAppearance` property using one of the thirteen text style attributes from `core/../values/type.xml`. Text can also be grouped if they are related and uses the same text style. This is done by creating a new attribute for the text group in `core/../values/attrs.xml` and assigning one of the thirteen text appearances to this new attribute. This attribute can then be set as the textAppearance for the group of texts. 
```

Comment:
this is also a bit of a deviation, we shouldn't define a new textAppearance attribute, the ones that exist come from materialComponents and 13 is enough of them. What I meant with a noContent style is that you can do this

```
<style = noContent> <item textAppearance=?headline1>
<-- some file-->
<TextView style=@style/noContent />
<-- another file-->
<TextView style=@style/noContent />
```

Basically a style allows us to define a rule "NoContentText has the text appearance Headline1"
Now I can alter this rule quite freely if I want NoContent to become Headline6, styles group views that should always display the same. We could refactor the UX and no content could become a lot less or a lot more prominent so it makes sense that we would only have to update the textAppearance in 1 place

Reply:
I was thinking a new style would override something important but see that it will not! Thanks again.